.github/workflows/labels_from_comment.yml

```diff
@@ -1,49 +1,72 @@
-# This workflow allows any user to add one of the `awaiting-author`, or `WIP` labels,
+# This workflow allows any user to add or remove one of the labels in the array `labelArray`
 # by commenting on the PR or issue.
-# Other labels from this set are removed automatically at the same time.
 
 name: Label PR based on Comment
 
 on:
   issue_comment:
     types: [created]
 
+# Limit permissions for GITHUB_TOKEN for the entire workflow
+permissions:
+  contents: read
+  pull-requests: write  # Only allow PR comments/labels
+  # All other permissions are implicitly 'none'
+
 jobs:
   update-label:
-    if: github.event.issue.pull_request && (contains(github.event.comment.body, 'awaiting-author') || contains(github.event.comment.body, 'WIP'))
+    env:
+      COMMENT: ${{ github.event.comment.body }}
     runs-on: ubuntu-latest
 
     steps:
     - name: Add / remove label based on comment
-      uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
-      with:
-        github-token: ${{ secrets.GITHUB_TOKEN }}
-        script: |
-          const { owner, repo, number: issue_number  } = context.issue;
-          const commentLines = context.payload.comment.body.split(/\r?\n/);
-
-          const awaitingAuthor = commentLines.includes('awaiting-author');
-          const wip = commentLines.includes('WIP');
-
-          const removeAwaitingAuthor = commentLines.includes('-awaiting-author');
-          const removeWip = commentLines.includes('-WIP');
-
-          if (removeAwaitingAuthor) {
-            await github.rest.issues.removeLabel({ owner, repo, issue_number, name: 'awaiting-author' }).catch(() => {});
-          }
-
-          if (removeWip) {
-            await github.rest.issues.removeLabel({ owner, repo, issue_number, name: 'WIP' }).catch(() => {});
-          }
-
-          if (awaitingAuthor || wip) {
-            await github.rest.issues.removeLabel({ owner, repo, issue_number, name: 'awaiting-author' }).catch(() => {});
-            await github.rest.issues.removeLabel({ owner, repo, issue_number, name: 'WIP' }).catch(() => {});
-          }
-
-          if (awaitingAuthor) {
-            await github.rest.issues.addLabels({ owner, repo, issue_number, labels: ['awaiting-author'] });
-          }
-          if (wip) {
-            await github.rest.issues.addLabels({ owner, repo, issue_number, labels: ['WIP'] });
-          }
+      run: |
+        labelArray=("awaiting-author" "WIP" "easy")
+
+        # we strip `\r` since line endings from GitHub contain this character
+        COMMENT="${COMMENT//$'\r'/}"
+
+        # trim leading/trailing whitespace and collapse "internal" whitespace
+        COMMENT="$(printf '%s' "${COMMENT}" | awk '{$1=$1};1')"
+
+        # If, after trimming whitespace, the comment spans more than one line, we do not act.
```

Comment:
We currently allow multiple actions in one comment -- are you planning to add that functionality back after more testing?

Reply:
What I was trying to do in the current workflow was to require that each "command" be on its own line, but I'm not sure how successfuly I was in actually implementing it. What you're suggesting might work too!